website/server/controllers/api-v3/challenges.js

```diff
@@ -410,13 +547,18 @@ api.exportChallengeCsv = {
 };
 
 /**
- * @api {put} /api/v3/challenges/:challengeId Update a challenge
+ * @api {put} /api/v3/challenges/:challengeId Update a challenge. Only reflects
+ * changes to the full name of the challenge. Other challenge details cannot be
+ * updated through this route.
```

Comment:
That is the correct behavior. You can't change the prize, group or tag name after creation.